mantidproject/mantid
# Sample material dialog

**Description of work.**

Reinstated the sample material dialog, which was present in Mantid Plot but not Workbench. It is accessed via an action on the context menu of the workspace widget, but only when a single workspace is selected. It has been rewritten as a python widget.

**To test:**

Load a workspace, right click on it, and click the "Sample material..." action in the context menu. It should display the values stored in the workspace's sample material. Use the "set" button to set the values of the sample material, and the "copy" button to copy the sample material to another workspace.
The context menu action should only appear if a single workspace is selected.
The conetxt menu action should NOT appear if the worksapce doesn't inherit from ExperimentInfo, because then it would not contain a sample object.

Fixes #30477. 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
